src/parser/deathknight/frost/modules/talents/BreathOfSindragosa.tsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+
+import { i18n } from '@lingui/core';
+import { t } from '@lingui/macro';
+
+import Analyzer, { SELECTED_PLAYER, Options } from 'parser/core/Analyzer';
+import { When, ThresholdStyle } from 'parser/core/ParseResults';
+import Events, {CastEvent, RemoveBuffEvent, FightEndEvent, DamageEvent, EventType, ClassResources, EnergizeEvent, Ability} from 'parser/core/Events';
+import EventEmitter from 'parser/core/modules/EventEmitter';
+import { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import Statistic from 'interface/statistics/Statistic';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+
+const GOOD_BREATH_DURATION_MS = 25000;
+const BREATH_COST_PER_TICK = 160;
+const HYPOTHERMIC_PRESENCE_COST_REDUCTION = .35;
+
+class BreathOfSindragosa extends Analyzer {
+  static dependencies = {
+    eventEmitter: EventEmitter,
+  }
+
+  protected eventEmitter!: EventEmitter;
+
+  damageAbility!: Ability;
+  triggeringCast!: CastEvent;
+  
+  beginTimestamp = 0;
+  casts = 0;
+  badCasts = 0;
+  totalDuration = 0;
+  breathActive = false;
+  mostRecentTickTime = 0;
+  currentRpAmount = 0;
+  
+
+  tickAmounts: number[] = [];
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.BREATH_OF_SINDRAGOSA_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.BREATH_OF_SINDRAGOSA_TALENT), this.onCast);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.BREATH_OF_SINDRAGOSA_TALENT), this.onRemoveBuff);
+    this.addEventListener(Events.fightend, this.onFightEnd);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.BREATH_OF_SINDRAGOSA_TALENT_DAMAGE_TICK), this.onDamage)
+    this.addEventListener(Events.energize.by(SELECTED_PLAYER), this.onEnergize);
+  }
+
+  onEnergize(event: EnergizeEvent) {
+    if(!this.breathActive || !event.classResources) {
+      return;
+    }
+    event.classResources
+      .filter(resource => resource.type === RESOURCE_TYPES.RUNIC_POWER.id)
+      .forEach(({ amount }) => {
+        this.currentRpAmount = amount;
+      });
+  }
+
+  onDamage(event: DamageEvent) {
+    if (event.timestamp === this.mostRecentTickTime) {
+      return;
+    }
+
+    if (!this.damageAbility) {
+      this.damageAbility = event.ability;
+    }
+
+    this.tickAmounts.push(this.selectedCombatant.hasBuff(SPELLS.HYPOTHERMIC_PRESENCE_TALENT.id, event.timestamp) ? BREATH_COST_PER_TICK * (1 - HYPOTHERMIC_PRESENCE_COST_REDUCTION) : BREATH_COST_PER_TICK);
```

Comment:
It may or may not be, my next task is to add a module for Hypothermic Presence, which grants a buff that reduces the cost of your RP spenders.  Because RP spent on BoS isn't logged I'm going to need to do something to pull out specifically which ticks of Breath are at a reduced cost.  I won't be surprised if I end up ripping out half of this when I start on that task.  Maybe I fabricate resourcespent events...